code/clue.js

```diff
@@ -1,59 +1,464 @@
-}
+killerCard.onclick = killerAnimation;
+
+// Actions related to Weapon deck
+
+const favouriteWeaponAnimation = () => {
+  if (mystery.killer === undefined) {
+    revealedMystery.innerHTML = ('Pick a killer first!');
+  } else {
+    weaponLoader.style.display = "block";
+    weaponName.innerHTML = '';
+    weaponWeight.innerHTML = '';
+    weaponImage.src = '';
+    weaponImage.style.display = "none";
+    weaponFavouriteRoom.innerHTML = '';
+    revealedMystery.innerHTML = '';
+    setTimeout(function () {
+      pickFavouriteWeapon(mystery.killer);
+    }, 2000);
+  };
+};
+
+const favouriteWeaponDuplicated = (elementToPopulate, howManyTimes) => {
+  let result = [];
+  for (let i = 0; i < howManyTimes; i++) {
+    result.push(elementToPopulate);
+  };
+  return result;
+};
+
+const pickFavouriteWeapon = (suspect) => {
+  const newArrayOfWeapons = weapons.map(a => a.name);
+  const randomWeapons = [
+    ...newArrayOfWeapons,
+    ...favouriteWeaponDuplicated(suspect.favouriteWeapon, 10)
+  ];
+  mystery.pickedWeapon = randomSelector(randomWeapons);
+  const mysteryWeapon = weapons.find((item) => {
+    if (item.name === mystery.pickedWeapon) {
+      return item;
+    };
+  });
+  mystery.weapon = mysteryWeapon;
+  mystery.favouriteRoom = mysteryWeapon.favouriteRoom;
+  weaponLoader.style.display = "none";
+  weaponCard.style.background = "#808080";
+  weaponName.innerHTML = mystery.weapon.name;
+  weaponWeight.innerHTML = `Weight: ${mystery.weapon.weight} oz`;
+  weaponImage.style.display = "inline-block";
+  weaponImage.src = mystery.weapon.image;
+  weaponFavouriteRoom.innerHTML = `Location: ${mystery.weapon.favouriteRoom}`;
+};
+
+weaponCard.onclick = favouriteWeaponAnimation;
+
+// Actions related to Room deck
+
+const favouriteRoomAnimation = () => {
+  if (mystery.killer === undefined) {
+    revealedMystery.innerHTML = ('Pick a killer first!');
+  } else if (mystery.weapon === undefined) {
+    revealedMystery.innerHTML = ('Pick a weapon first!');
+  } else {
+    roomLoader.style.display = "block";
+    roomName.innerHTML = '';
+    roomImage.src = '';
+    roomImage.style.display = "none";
+    setTimeout(function () {
+      pickFavouriteRoom(mystery.weapon);
+    }, 2000);
+  };
+};
+
+const favouriteRoomDuplicated = (elementToPopulate, howManyTimes) => {
+  let result = [];
+  for (let i = 0; i < howManyTimes; i++) {
+    result.push(elementToPopulate);
+  };
+  return result;
+};
+
+const pickFavouriteRoom = (weapon) => {
+  const newArrayOfRooms = rooms.map(a => a.name);
+  console.log(newArrayOfRooms);
+  const randomRooms = [
+    ...newArrayOfRooms,
+    ...favouriteRoomDuplicated(weapon.favouriteRoom, 30)
+  ];
+  console.log(favouriteRoomDuplicated(weapon.favouriteRoom, 30));
+  console.log(randomRooms);
+  mystery.pickedRoom = randomSelector(randomRooms);
+  const mysteryRoom = rooms.find((item) => {
+    if (item.name === mystery.pickedRoom) {
+      return item;
+    };
+  });
+  console.log(mysteryRoom);
```

Comment:
console.logs are awesome in development but not to be left behind 😇

Reply:
Argh! I deleted the ones I commented out and left the visible ones! Thanks for pointing that out, I'm gonna take care of them 🚮